airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableRow.tsx

```diff
@@ -0,0 +1,147 @@
+import { StreamHeaderProps } from "../StreamHeader";
+import { HeaderCell } from "../styles";
+import styles from "./CatalogTreeTableRow.module.scss";
+
+export const CatalogTreeTableRow: React.FC<StreamHeaderProps> = ({
+  stream,
+  destName,
+  destNamespace,
+  // onSelectSyncMode,
+  onSelectStream,
+  // availableSyncModes,
+  pkType,
+  onPrimaryKeyChange,
+  onCursorChange,
+  primitiveFields,
+  cursorType,
+  // isRowExpanded,
+  fields,
+  onExpand,
+  changedSelected,
+  hasError,
+  disabled,
+}) => {
+  const { primaryKey, syncMode, cursorField, destinationSyncMode } = stream.config ?? {};
+  const isStreamEnabled = stream.config?.selected;
+
+  const { defaultCursorField } = stream.stream ?? {};
+  const syncSchema = useMemo(
+    () => ({
+      syncMode,
+      destinationSyncMode,
+    }),
+    [syncMode, destinationSyncMode]
+  );
+
+  const [isSelected, selectForBulkEdit] = useBulkEditSelect(stream.id);
+
+  const paths = useMemo(() => primitiveFields.map((field) => field.path), [primitiveFields]);
+  const fieldCount = fields?.length ?? 0;
+  const onRowClick = fieldCount > 0 ? () => onExpand() : undefined;
+
+  const iconStyle = classnames(styles.icon, {
+    [styles.plus]: isStreamEnabled,
+    [styles.minus]: !isStreamEnabled,
+  });
+
+  const streamHeaderContentStyle = classnames(styles.streamHeaderContent, {
+    [styles.enabledChange]: changedSelected && isStreamEnabled,
+    [styles.disabledChange]: changedSelected && !isStreamEnabled,
+    [styles.selected]: isSelected,
+    [styles.error]: hasError,
+  });
+
+  const checkboxCellCustomStyle = classnames(styles.checkboxCell, styles.streamRowCheckboxCell);
+
+  return (
+    <Row onClick={onRowClick} className={streamHeaderContentStyle}>
+      {!disabled && (
+        <div className={checkboxCellCustomStyle}>
+          {changedSelected && (
+            <div>
+              {isStreamEnabled ? (
+                <FontAwesomeIcon icon={faPlus} size="2x" className={iconStyle} />
+              ) : (
+                <FontAwesomeIcon icon={faMinus} size="2x" className={iconStyle} />
+              )}
+            </div>
+          )}
+          <CheckBox checked={isSelected} onChange={selectForBulkEdit} />
+        </div>
+      )}
+      <Cell>
+        <Switch small checked={stream.config?.selected} onChange={onSelectStream} disabled={disabled} />
+      </Cell>
+      <Cell>{fieldCount}</Cell>
+      <HeaderCell ellipsis title={stream.stream?.namespace || ""}>
+        {stream.stream?.namespace || <FormattedMessage id="form.noNamespace" />}
+      </HeaderCell>
+      <Cell>{stream.stream?.name}</Cell>
+      <Cell>
+        {disabled ? (
+          <HeaderCell ellipsis title={syncSchema.syncMode}>
+            {syncSchema.syncMode}
+          </HeaderCell>
+        ) : (
+          // TODO: Replace with Dropdown/Popout
+          syncSchema.syncMode
+        )}
+      </Cell>
+      <HeaderCell>
+        {cursorType && (
+          <PathPopout
```

Comment:
These dropdowns aren't stopping click event propagation, so clicking on one unexpectedly opened the streams panel.  Though, checking the Figma prototype, that's what happens there too.  The dropdown caret makes me think that it will be a dropdown, not open the entire panel.

Reply:
Yes, this is something that I noticed but decided not to fix because these components are temporary This is one of those areas that I expect to be changed as we work more on this component.